test/variable.jl

```diff
@@ -673,3 +674,97 @@ end
         "Julia's broadcast syntax instead: `JuMP.value.(x)`.")
     @test_throws exception JuMP.value(x)
 end
+
+struct MockModel <: JuMP.AbstractModel
+    has_duals::Bool
+    obj_sense::MathOptInterface.OptimizationSense
+end
```

Comment:
Take a look at mock optimizer in https://github.com/JuliaOpt/JuMP.jl/blob/master/test/generate_and_solve.jl